hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -1410,6 +1417,13 @@ private long gasFeeInTinybars(final TransactionBody.Builder txBody, final Instan
 		return fees.getServiceFee() + fees.getNetworkFee() + fees.getNodeFee();
 	}
 
+	private void validateAndSetRevertReasonIfNeeded(final boolean flag, final String revertMessage) {
+		if (!flag) {
+			messageFrame.setRevertReason(Bytes.of(revertMessage.getBytes()));
+			throw new InvalidTransactionException(FAIL_INVALID);
```

Comment:
We want to reserve the `FAIL_INVALID` response for an internal server error (i.e. a bug in our code).

But the method is being called if the user passes an invalid mint/burn amount; so we should choose/create a more descriptive response code here.

Reply:
In the protobufs the amount type for mint and burn is defined as uint64, which has a max value of 2^63-1 (the same as long).
In solidity the uint64 type, however, has a max value of 2^64-1 and this value won't fit in the protobuf field.
That's why we use precompile argument as BigInteger for these fields, so that we would be able to properly decode any value above 2^63-1 and before setting it to the protobuf field to validate it and set a specific error message.